my-webxr-app/src/components/DataPoint.tsx

```diff
@@ -43,17 +46,37 @@ export default function DataPoint({
     setHoverCount(amount);
   };
 
+  // Display the point characteristics in console
+  const showSelectedDataPointInfo = (id: number, marker: string, color: string, column1: string, column2: string, column3: number, meshProps?: MeshProps ) => {
+    console.log("Selected Data Point:");
+    console.log("ID:", id);
+    console.log("Marker:", marker);
+    console.log("Color:", color);
+    console.log("Position:", meshProps?.position);
+    console.log("Column 1:", column1);
+    console.log("Column 2:", column2);
+    console.log("Column 3:", column3);
+  }
+
   return (
     <Interactive
       onHover={() => adjustHoverCount(hoverCount + 1)}
       onBlur={() => adjustHoverCount(hoverCount - 1)}
       onSelect={() => {
-        selectedDataPoint === id
-          ? setSelectedDataPoint(null)
-          : setSelectedDataPoint(id);
+        // Clear current selection
+        if (selectedDataPoint?.id === id) {
+          setSelectedDataPoint(null);
+        }
+        // Update new selected point with its characteristics and print to UI
+        else {
+          setSelectedDataPoint({id, marker, color, column1, column2, column3, meshProps});
+          console.log(selectedDataPoint?.meshProps?.position)
+          showSelectedDataPointInfo(id, marker, color, column1, column2, column3, meshProps);
+        }
```

Comment:
if the point is already selected, remove the selection

else, select the point.

Reply:
Yes, that's correct! I will specify that more clearly in my comments.